rotkehlchen/globaldb/handler.py

```diff
@@ -208,6 +209,140 @@ def add_asset(
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(
+                'SELECT assets.identifier, assets.name, common_asset_details.symbol FROM assets '
+                'JOIN common_asset_details on assets.identifier = common_asset_details.identifier '
+                'WHERE ' + identifiers_query,
+                tuple(identifiers),
+            )
+            for entry in cursor:
+                result.append({
+                    'identifier': entry[0],
+                    'name': entry[1],
+                    'symbol': entry[2],
+                })
+            if len(result) != len(identifiers):
+                raise InputError('One or more identifiers do not exist')
+        return result
+
+    @staticmethod
+    def search_assets(filter_query: 'AssetsFilterQuery') -> List[Dict[str, str]]:
+        search_result = []
+        query, bindings = filter_query.prepare()
+        parent_query = """
+        SELECT assets.identifier, name, symbol FROM assets
+        JOIN common_asset_details on assets.identifier = common_asset_details.identifier
+        """
+        query = parent_query + query
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(query, bindings)
+            for entry in cursor:
+                search_result.append({
+                    'identifier': entry[0],
+                    'name': entry[1],
+                    'symbol': entry[2],
+                })
+        return search_result
+
+    @staticmethod
+    def search_assets_levenshtein(
+            filter_query: 'AssetsFilterQuery',
+            substring_search: str,
+            limit: Optional[int],
+    ) -> List[Dict[str, str]]:
+        search_result = []
+        query, bindings = filter_query.prepare()
+        parent_query = """
+        SELECT assets.identifier, name, symbol FROM assets
+        JOIN common_asset_details on assets.identifier = common_asset_details.identifier
+        """
+        query = parent_query + query
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(query, bindings)
+            for entry in cursor:
+                lev_dist_name = levenshtein_distance(substring_search, entry[0])
+                lev_dist_symbol = levenshtein_distance(substring_search, entry[1])
+                lev_dist_average = (lev_dist_name + lev_dist_symbol) / 2
+                if lev_dist_average <= 4:
+                    search_result.append({
+                        'identifier': entry[0],
+                        'name': entry[1],
+                        'symbol': entry[2],
+                        'lev_dist': lev_dist_average,
+                    })
+        search_result = sorted(search_result, key=lambda item: item['lev_dist'])
+        _ = [result.pop('lev_dist') for result in search_result]  # remove the unneeded key. negligible time complexity  # noqa: E501
```

Comment:
maybe do it in a normal for loop? It looks strange with the list comprehension as it is not the kind of thing you do with them + you are creating a new object with `len(search_result)` elements